src/cargo/core/workspace.rs

```diff
@@ -561,6 +561,13 @@ impl<'cfg> Workspace<'cfg> {
         self.member_ids.contains(&pkg.package_id())
     }
 
+    /// Returns true if this workspace is being used for a `cargo install` operation
+    pub fn is_install(&self) -> bool {
+        // For now, this condition detects `cargo install`, but calling `is_install` allows us to
+        // more easily change that in the future.
+        self.is_ephemeral && self.ignore_lock
```

Comment:
`cargo install --locked` is still an install but wouldn't this function return `false`?

Reply:
Oh, good point. I did this based on inspiration from https://github.com/rust-lang/cargo/pull/10891#discussion_r941806128 . @ehuss, is there a variant of this that would make that code clearer, or should I just drop this change?

Reply:
I don't think there is a good way to detect if it is an `install`.  Historically we have tried to avoid making that kind of information available since it is an abstraction loss (the lower-level build code shouldn't know about higher-level concepts like "install").

I think it might be ok to add something to `BuildConfig` to indicate which command initiated a build. The risk is that it will be abused for changing behavior based on the command. Perhaps if there is a stern comment emphasizing it should only be used for diagnostics would help?

I wouldn't make this a property of the workspace. I forgot to also say that `is_ephemeral` is only `true` when installing something from a registry (not a local or git install).